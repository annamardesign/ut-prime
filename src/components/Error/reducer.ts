const defaultErrors = {
    401: 'Session closed or expired',
    403: 'Insufficient permissions for the operation',
    404: 'Resource cannot be found',
    500: 'Error was received from server. Please try again later'
};

const mapErrorMessage = (resp) => {
    let returnMsg = resp.print || (resp.statusCode && defaultErrors[resp.statusCode]) || 'Unexpected error';
    if (resp.validation && resp.validation.keys && resp.validation.keys.length > 0) {
        returnMsg = resp.validation.keys.reduce((prev, cur) => {
            prev.push(cur);
            return prev;
        }, ['Validation error in:']);
        return returnMsg.join(' ');
    }
    return returnMsg;
};

export default (state = {open: false, title: '', message: '', type: ''}, action) => {
    if (action.suppressErrorWindow) return state;
    if (action.type === 'front.error.close') {
        return {
            ...state,
            open: false,
            title: '',
            message: '',
            statusCode: 200,
            type: ''
        };
    }
    if (action.error) {
        const msg = (mapErrorMessage(action.error) || mapErrorMessage(state.message));
        const statusMessage = action.error.statusMessage;
        const statusCode = action.error.statusCode;
        let title = 'Error';
        const type = action.error.type;
        if (statusMessage) {
            title = `${statusMessage}`;
            if (statusCode) {
                title += `(${statusCode})`;
            }
        }

        return {
            ...state,
            open: true,
            title: title,
            message: msg,
            statusCode,
            type
        };
    }
    return state;
};
