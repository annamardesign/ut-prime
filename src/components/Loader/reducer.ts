import immutable from 'immutable';

const defaultState = immutable.fromJS({
    open: false,
    requests: 0,
    message: 'Loading, please wait...'
});

export default (state = defaultState, action) => {
    if (action.suppressPreloadWindow) {
        return state;
    }
    if (action.methodRequestState === 'requested') {
        return state
            .set('open', true)
            .set('requests', state.get('requests') + 1)
            .set('message', action.prefetchWindowText || defaultState.get('message'));
    } else if (action.methodRequestState === 'finished') {
        const requests = state.get('requests');
        if (requests <= 1) {
            return state
                .set('open', false)
                .set('requests', 0);
        } else {
            return state
                .set('requests', requests - 1);
        }
    }
    return state;
};
