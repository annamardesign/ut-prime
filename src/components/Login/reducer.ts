import {LOGIN, COOKIE_CHECK} from './actions';
import Immutable from 'immutable';

export default (state = null, action: { type: Symbol; methodRequestState: string; result: {language: {}}; }) => {
    if ([LOGIN, COOKIE_CHECK].includes(action.type) && action.methodRequestState === 'finished') {
        //! todo simplify
        return action.result ? Immutable.fromJS({
            result: action.result,
            language: action.result.language
        }) : false;
    }
    return state;
};
