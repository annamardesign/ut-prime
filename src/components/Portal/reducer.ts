const handlers = {
    'front.tab.show'(state, {title, path, component}) {
        return {
            ...state,
            tabs: [...(state.tabs || []), {
                title: title,
                path: path,
                component
            }]
        };
    },
    'front.tab.close'(state, {index}) {
        return {
            ...state,
            tabs: (items => {
                items.splice(index, 1);
                return items;
            })([...(state.tabs || [])])
        };
    }
};

export default (state = {}, action) => {
    const handler = handlers[action.type];
    return handler ? handler(state, action) : state;
};
