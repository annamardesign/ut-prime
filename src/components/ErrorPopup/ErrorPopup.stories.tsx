import React from 'react';
import { withReadme } from 'storybook-readme';
import { action } from '@storybook/addon-actions';

// @ts-ignore: md file and not a module
import README from './README.md';
import ErrorPopup from './index';

export default {
    title: 'ErrorPopup',
    component: ErrorPopup,
    decorators: [withReadme(README)]
};

export const Basic: React.FC<{}> = () => <ErrorPopup onClick={action('clicked')}>Hello ErrorPopup</ErrorPopup>;
