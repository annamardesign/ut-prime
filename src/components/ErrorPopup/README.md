# Button

## How to use

```jsx
import ErrorPopup from 'ut-front-devextreme/core/ErrorPopup';

<ErrorPopup onClick={handlerOnClick}>Hello ErrorPopup</ErrorPopup>;
```

## Props

- **className** - class applied to root `ErrorPopup` html element.

| propName  | propType | defaultValue | isRequired |
| --------- | -------- | ------------ | ---------- |
| className | string   |              | no         |
