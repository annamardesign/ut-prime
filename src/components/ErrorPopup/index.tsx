import React from 'react';
import clsx from 'clsx';
import { connect } from 'react-redux';
import { Popup } from 'devextreme-react/popup';
import { Button } from 'devextreme-react/button';

import Text from '../Text';
import { logout } from '../login/actions';

import { Styled, StyledType } from './ErrorPopup.types';

const close = () => ({type: 'error.close'});

const ErrorPopup: StyledType = ({ classes, open, message, close, title, type, logout }) => {
    let handleClose = close;
    let showCloseButton = true;
    let actionButtons;

    if (type === 'identity.invalidCredentials') {
        handleClose = logout;
        showCloseButton = false;
        actionButtons = <Button
            width={120}
            text="Login"
            type="default"
            stylingMode="contained"
            onClick={handleClose}
        />;
    } else {
        actionButtons = <Button
            width={120}
            text="Close"
            type="default"
            stylingMode="contained"
            onClick={handleClose}
        />;
    }

    return (
        <Popup
            visible={open}
            minWidth={540}
            width='auto'
            height='auto'
            onHidden={handleClose}
            {...{showCloseButton, title}}
        >
            <div className={clsx(classes.errorIconWrap, classes.errorIcon)} />
            <div className={classes.errorMessageWrap}>
                <Text>{message}</Text>
            </div>
            <div className={classes.errorButtonWrap}>
                {actionButtons}
            </div>
        </Popup>
    );
};

export default connect(
    state => state.errorPopup,
    {close, logout}
)(Styled(ErrorPopup));
