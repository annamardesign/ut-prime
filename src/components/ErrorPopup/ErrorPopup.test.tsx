import React from 'react';

import { render } from '../test';
import ErrorPopup from './index';
import reducer from './reducer';

describe('<ErrorPopup />', () => {
    it('render equals snapshot', () => {
        const { getByTestId } = render(<ErrorPopup />, {errorPopup: {open: true}}, reducer);
        expect(getByTestId('ut-front-test')).toMatchSnapshot();
    });
});
