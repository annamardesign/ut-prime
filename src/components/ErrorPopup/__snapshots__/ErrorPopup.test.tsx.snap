// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPopup /> render equals snapshot 1`] = `
<div
  data-testid="ut-front-test"
>
  <div
    class="dx-overlay dx-popup dx-widget dx-visibility-change-handler"
  >
    <div
      class="dx-overlay-content dx-popup-fullscreen-width dx-popup-normal"
      style="min-width: 540px; width: auto; height: auto;"
      tabindex="0"
    >
      <div
        class="dx-popup-content"
      />
    </div>
  </div>
</div>
`;
